lib/bundler/diff.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Bundler
+  class Diff
+    def initialize(string1, string2)
+      @string1 = string1
+      @string2 = string2
+    end
+
+    def diff
+      @diff ||= begin
+          @paths = [tempfile(@string1), tempfile(@string2)]
+
+          cmd = format '"%s" %s %s', diff_bin, "-U10000", @paths.map {|s| %("#{s}") }.join(" ")
+          diff = `#{cmd}`
+          diff.force_encoding("ASCII-8BIT") if diff.respond_to?(:valid_encoding?) && !diff.valid_encoding?
```

Comment:
This sounds like old times legacy code. I would remove it.

Reply:
I agree that this seems like code intended to support older versions of Ruby. Can you either explain what this line is needed for in Ruby 2.3+ or remove it?